packages/legend-application-query/src/stores/QueryEditorStore.ts

```diff
@@ -85,9 +84,78 @@ import {
   ClassQueryBuilderState,
   MappingQueryBuilderState,
   ServiceQueryBuilderState,
+  QueryLoaderState,
 } from '@finos/legend-query-builder';
+import {
+  LEGEND_QUERY_USER_DATA_KEY,
+  USER_DATA_RECENTLY_VIEWED_QUERIES_LIMIT,
+} from '../application/LegendQueryUserData.js';
+import { createModelSchema, list, primitive } from 'serializr';
 import { LegendQueryTelemetryHelper } from '../__lib__/LegendQueryTelemetryHelper.js';
 
+export class QueryStorageState {
```

Comment:
don't really have a better way to suggest, but I felt like this could have been organized slightly better, potentially like how we organize `application` settings for example